### DISCUSSION TITLE
How do I use multi gpu setups?

### DISCUSSION THREAD

**Discussion by FileDotZip** [ggml-org/llama.cpp#2132] [General]
I have a intel scalable gpu server, with 6x Nvidia P40 video cards with 24GB of VRAM each.

How can I specify for llama.cpp to use as much vram as it needs from this cluster of gpu's? Does it automatically do it?

I am following [this guide](https://blog.gopenai.com/the-total-noobs-guide-to-harnessing-the-gpu-for-llama-inference-2e3c4fdfbc84) at step 6

**Comment by JohannesGaessler** [comment]
https://github.com/ggerganov/llama.cpp/tree/master/examples/main
